DSpace/dspace-replicate
# Update BagIt packaging

Changes for: https://jira.lyrasis.org/browse/DS-4447

Tests will be incorporated in another PR to keep the changes here easier to review.

* Adds bagit-support library to handle Bagging and BagProfile
* Create BagItAipWriter to keep track of state when packaging Bags
* Creates two helper classes to help encapsulate information
  * BagBitstream stores the bitstream, parent bundle name, and xml elements for the bitstream
  * XmlElement stores attributes and a body for an xml writer
* Updated packers (Item, Collection, Community, Catalog) to be scoped around collecting data to be written to the Bag
* Drops support for fetch.txt in BagIt packaging

------------------------------------------
Testing:
1. Create a new DSpace repository with the RTS enabled as an addon
2. Create a Community, Collection, and Item in the repository
   1. Optionally - add a logo, or other information for the given object
3. Under the curation context for the Community, select the `Transfer AIP to storage` task
4. Review the 3 AIPs which were written to the repstore directory of your dspace installation
   1. Each AIP should be in a zip format
   2. Each AIP should have all present members (object.properties, metadata.xml, bitstreams)

@tdonohue @bbranan
